### PR TITLE
Export additional metrics to Prometheus

### LINKED ISSUE #13921: [monitoring] Prometheus metrics export missing disk and network utilization metrics
### Describe your feature request

I'm following this guide https://docs.ray.io/en/master/ray-metrics.html to export Ray cluster metrics to Prometheus for monitoring. I was able to get metrics like `ray_node_cpu`, `ray_node_mem`, which are great, but I'm also looking for disk utilization, network bytes sent/received, and network speed. These metrics are available on the Ray dashboard GUI, so I'm guessing it's a matter of exporting them to Prometheus.

Also I noticed the memory usage is reported in MB, not bytes. The latter is recommended by Prometheus best practice: https://prometheus.io/docs/practices/naming/.

### PR DESCRIPTION
**Pull request by kathryn-zhou**
## Why are these changes needed?

Exports additional Ray metrics to Prometheus. The metrics added are: disk usage, disk utilization, network bytes (sent/received), and network speed (send/receive)

## Related issue number

Closes #13921 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/reporter/reporter_agent.py [hunk 1]
```diff
@@ -237,8 +254,10 @@ def _get_all_stats(self):
         self._network_stats_hist.append((now, network_stats))
         self._network_stats_hist = self._network_stats_hist[-7:]
         then, prev_network_stats = self._network_stats_hist[0]
-        netstats = ((network_stats[0] - prev_network_stats[0]) / (now - then),
-                    (network_stats[1] - prev_network_stats[1]) / (now - then))
+        net_speed_stats = ((network_stats[0] - prev_network_stats[0]) /
```
Comment (simon-mo) [inline dashboard/modules/reporter/reporter_agent.py:257]:
```suggestion
        network_speed_stats = ((network_stats[0] - prev_network_stats[0]) /
```
The code style for Ray OSS is to use full spelling if possible and try to be consistent with surrounding code. So net -> network

Comment (simon-mo) [reply dashboard/modules/reporter/reporter_agent.py:257]: To make it easier to read, let's unwind the [0]s and [1]s. 
```
prev_send, prev_recv = prev_network_stats
now_send, now_recv = network_stats
duration = now- then
net_speed_stats = (now_send-prev_send)/duration, (now_recv-prev_recv)/duration
```

Comment (kathryn-zhou) [reply dashboard/modules/reporter/reporter_agent.py]: made changes

**Review by simon-mo — APPROVED** [review]

File: dashboard/modules/reporter/reporter_agent.py [hunk 2]
```diff
@@ -77,7 +77,25 @@ def __init__(self, dashboard_agent):
...
+                              "bytes", ["ip"]),
+            "node_disk_usage": Gauge("node_disk_usage",
+                                     "Total disk usage (bytes) on a ray node",
+                                     "bytes", ["ip"]),
+            "node_disk_utilization_percentage": Gauge(
+                "node_disk_utilization_percentage",
+                "Total disk utilization (percentage) on a ray node",
+                "percentage", ["ip"]),
+            "node_network_sent": Gauge("node_network_sent",
+                                       "Total network sent", "bytes", ["ip"]),
```
Comment (edoakes) [inline dashboard/modules/reporter/reporter_agent.py:89]: bytes/s is a weird unit for network util. Usually you use bits (likely Kb in this case). If this is what we're showing in the dashboard, it's fine for now

**Review by edoakes — APPROVED** [review]

**Pull request merged**